### PR TITLE
Fixes #31791 - Add duplicate name validation to bookmarks

### PR DESCRIPTION
**Pull request by amirfefer**
Saving a bookmark with an existing/duplicate bookmark name doesn't show a clear error.

### CODE REVIEW

File: webpack/assets/javascripts/react_app/components/Bookmarks/BookmarksActions.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ export const getBookmarks = (url, controller) => {
   const uri = new URI(url);
 
   // eslint-disable-next-line camelcase
-  uri.setSearch({ search: `controller=${controller}`, per_page: 100 });
+  uri.setSearch({ search: `controller=${controller}`, per_page: 'all' });
```
Comment (amirfefer) [inline webpack/assets/javascripts/react_app/components/Bookmarks/BookmarksActions.js:16]: There's no pagination in the bookmarks dropdown

**Comment by xprazak2** [comment]
Works, test failures related.

I have one additional concern: You take the advantage of parent component having bookmarks already accessible and introducing client side uniqueness check works pretty well here as users do not need to submit the form to see the errors. Though I am pretty sure the server side errors used to be displayed correctly, and now I can observe the uncaught SumbissionError in the console, so perhaps some part of how we handle error responses has changed. @amirfefer do you think it is worth investing time in tracking that down?

bookmarks-taken

**Comment by amirfefer** [comment]
Thanks, @xprazak2! 
I've created https://github.com/theforeman/foreman/pull/8309 for fixing error displaying in the toast notification, this error affects also the setting's form modal.

This PR can be kept for handling bookmarks client-side validation.

**Review by xprazak2 — APPROVED** [review]

**Comment by xprazak2** [comment]
Thanks!

**Comment by ezr-ondrej** [comment]
This probably should get cherry-picked, right? If I understand correctly it is a regression.

**Comment by ezr-ondrej** [comment]
CP 2.4: #8506

**Pull request merged**